js/lazy_images.js

```diff
@@ -27,98 +32,100 @@ function setImgLazyLoad(result) {
-      rootMargin: "0px",
-      threshold: 0.5
-    };
-
-    // register the config object with an instance
-    // of intersectionObserver
-    var obst = new IntersectionObserver(function (entries, self) {
-      // iterate over each entry
-      entries.forEach(entry => {
-        // process just the images that are intersecting.
-        // isIntersecting is a property exposed by the interface
-        if (entry.isIntersecting) {
-          // custom function that copies the path to the img
-          // from data-src to src
-          entry.target.src = entry.target.getAttribute(
-            "data-src-fasterpageload"
-          );
-          entry.target.removeAttribute("data-src-fasterpageload");
-          console.log(
-            "[faster pageload plugin] Lazy loaded image: ",
-            entry.target.src
-          );
-          // the image is now in place, stop watching
-          self.unobserve(entry.target);
-        }
-      });
-    }, config);
-
-    var imgElements = document.querySelectorAll("img");
-    console.log("[faster pageload plugin] Lazyload " + imgElements.length + " images");
-
-    var foundSrcs = {}
-    var foundTimes = 0
-    var ourLazy = true;
-
-    for (var i = imgElements.length - 1; i >= 0; i--) {
-      var imgElem = imgElements[i];
-    
-
-      // if (
-      //   imgElem.src != undefined &&
-      //   imgElem.src != "" &&
-      //   Object.keys(imgElem.dataset).length == 0 &&
-      //   imgElem.src.includes("data:") == false
-      // ) {
-
-        if(foundSrcs[imgElem.src]== undefined){
-          foundSrcs[imgElem.src] = 1;
-        }
-        foundSrcs[imgElem.src]++;
-
-        console.log("FOUND: ",foundSrcs[imgElem.src], imgElem.src)
-
-        if (foundSrcs[imgElem.src] >= 5) {
-          console.log("[faster pageload plugin] Unknown lazy load plugin found. Remove lazy load elements: ", imgElements.length - 1 - i)
-          // we have apparently an unknown lazy load plugin on the site. Reset all images
-          for (var j = imgElements.length - 1; j >= i; j--) {
-            var remElem = imgElements[j];
-            remElem.src = remElem.getAttribute(
-              "data-src-fasterpageload"
-            );
-            remElem.removeAttribute("data-src-fasterpageload");
-            obst.unobserve(remElem);
-          }
-          ourLazy = false;
-          break
-
-        } else {
-
-        
-
-
-          if (imgElem.loading != undefined) {
-            imgElem.loading = "lazy"
-          } else {
-            imgElem.setAttribute("data-src-fasterpageload", imgElem.src);
-            imgElem.src = chrome.runtime.getURL("static/rocket.svg");
-            obst.observe(imgElem);
-          }
-        }
-     // }
-
-    }
-
-
-
-    browser.runtime.sendMessage({
-      lazyDone: true,
-      ourLazyLoad:   ourLazy
-    });
+	  if (!(arrayContains(window.location.host, result.blacklistSites.split(", ")))) {
```

Comment:
Please use the [happy path scheme](https://en.wikipedia.org/wiki/Happy_path) as it gives less indentation and creates a better to read code.

```
if (!(result.imgLazyLoad == true) || arrayContains(window.location.host, result.blacklistSites.split(", "))) { 
  return
}
```

Reply:
Done

Reply:
Ah... I did them in separate if statements - is that fine or do they need to be in one?

Reply:
That's fine. Just different coding styles. maybe two is even better understandable